rfcs/0035-create-connection.md

```diff
@@ -0,0 +1,143 @@
+---
+feature: Create Connection
+authors:
+  - "Bohan Zhang"
+start_date: "2023/01/15"
+---
+
+# RFC: Create Connection
+
+## Motivation
+
+A connection is used to describe how to connect to an external system that users want to read data from. Once created, a connection is reusable across `CREATE SOURCE` and `CREATE SINK` statements. This RFC proposes a new `CREATE CONNECTION` statement to create a connection.
+
+## Design
+
+### Syntax
+
+```sql
+-- create connection
+CREATE CONNECTION (IF NOT EXIST) {{ connection_name }}
+  TO {{ connection_type }}
+  ( {{ field }} = {{ value }}, ...);
+
+-- create source with connection
+CREATE SOURCE {{ source_name }} ( {{ field }} = {{ value }}, ... )
+  FROM CONNECTION {{ connection_name }}
+  with ( {{ field }} = {{ value }}, ...);
+
+-- create sink with connection
+CREATE SINK {{ sink_name }} ( {{ field }} = {{ value }}, ... )
+  FROM {{ source_name }}
+  INTO CONNECTION {{ connection_name }}
+  with ( {{ field }} = {{ value }}, ...);
+```
+
+If any field is specified both in `with` clause and `CREATE CONNECTION` statement, the value in `with` clause will be used.
+
+#### AWS Private Link
+
+An AWS Private Link establishes a private connection between a VPC and a service hosted on AWS. It is a secure and scalable way to connect to AWS services from within a VPC. The following fields are required to create an AWS Private Link connection:
```

Comment:
IIUC, Private Link is orthogonal with the service to connect, and it only makes sense when being used with the definition of source/sink together, right?

PS. I guess you borrowed the idea from https://materialize.com/docs/sql/create-connection/

Reply:
yes, private link provides a way to allow the service in our vpc to access resources in another vpc but it cannot be reversed.